pkg/stanza/fileconsumer/config.go

```diff
@@ -66,22 +65,22 @@ func NewConfig() *Config {
 // Config is the configuration of a file input operator
 type Config struct {
 	matcher.Criteria        `mapstructure:",squash"`
-	IncludeFileName         bool                     `mapstructure:"include_file_name,omitempty"`
-	IncludeFilePath         bool                     `mapstructure:"include_file_path,omitempty"`
-	IncludeFileNameResolved bool                     `mapstructure:"include_file_name_resolved,omitempty"`
-	IncludeFilePathResolved bool                     `mapstructure:"include_file_path_resolved,omitempty"`
-	PollInterval            time.Duration            `mapstructure:"poll_interval,omitempty"`
-	StartAt                 string                   `mapstructure:"start_at,omitempty"`
-	FingerprintSize         helper.ByteSize          `mapstructure:"fingerprint_size,omitempty"`
-	MaxLogSize              helper.ByteSize          `mapstructure:"max_log_size,omitempty"`
-	MaxConcurrentFiles      int                      `mapstructure:"max_concurrent_files,omitempty"`
-	MaxBatches              int                      `mapstructure:"max_batches,omitempty"`
-	DeleteAfterRead         bool                     `mapstructure:"delete_after_read,omitempty"`
-	Multiline               tokenize.MultilineConfig `mapstructure:"multiline,omitempty"`
-	TrimConfig              trim.Config              `mapstructure:",squash,omitempty"`
-	Encoding                string                   `mapstructure:"encoding,omitempty"`
-	FlushPeriod             time.Duration            `mapstructure:"force_flush_period,omitempty"`
-	Header                  *HeaderConfig            `mapstructure:"header,omitempty"`
+	IncludeFileName         bool            `mapstructure:"include_file_name,omitempty"`
+	IncludeFilePath         bool            `mapstructure:"include_file_path,omitempty"`
+	IncludeFileNameResolved bool            `mapstructure:"include_file_name_resolved,omitempty"`
+	IncludeFilePathResolved bool            `mapstructure:"include_file_path_resolved,omitempty"`
+	PollInterval            time.Duration   `mapstructure:"poll_interval,omitempty"`
+	StartAt                 string          `mapstructure:"start_at,omitempty"`
+	FingerprintSize         helper.ByteSize `mapstructure:"fingerprint_size,omitempty"`
+	MaxLogSize              helper.ByteSize `mapstructure:"max_log_size,omitempty"`
+	MaxConcurrentFiles      int             `mapstructure:"max_concurrent_files,omitempty"`
+	MaxBatches              int             `mapstructure:"max_batches,omitempty"`
+	DeleteAfterRead         bool            `mapstructure:"delete_after_read,omitempty"`
+	SplitConfig             split.Config    `mapstructure:"multiline,omitempty"`
```

Comment:
We are not going to change how it's exposed to end-user as `multiline`, right?

Is there any other use cases where this config is being used?

Reply:
Right, this doesn't change the user facing config at all.

> Is there any other use cases where this config is being used? I mean where it's actually used for splitting rather then multiline which is rather combining?

I'm not aware of any other uses outside of pkg/stanza. 

I think the word "multiline" is a bit of a misnomer. What is really meant here is "splitting based on something other than newlines". This can mean splitting out multiple logs from one line, or reading in a single log that spans multiple lines. Either way, it is concerned more with splitting than combining. (The recombine operator is more directly meant for "combining" logs.)

Reply:
Ok, I'm wondering if we can pick a word that works for splitting and combining so it can be exposed to the user with the same word but nothing good comes to my mind

Reply:
Hmm, not sure I agree that we should indicate combining here either way. It's all about splitting, just a question of whether we split on newlines or something else.

In terms of the actual term exposed to users in the config, I think multiline is not ideal, but in my experience it's been much more common to find logs that span multiple lines vs having multiple logs in a single line, so multiline is at least descriptive of the more common case. And of course logs that are naturally separated by newlines don't require configuration of the multiline section at all.